### PR TITLE
Updates to Migrating C++/CLI projects to .NET Core and .NET 5+

### LINKED ISSUE #33536: Doc needs to mention that C++/CLI .NET exe is not supported.
To migrate existing C++/CLI .NET Framework exe people need to create a c# .NET exe project which reference C++/CLI .NET dll.

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: 590dc11f-354e-ded2-469f-605febf6b394
* Version Independent ID: 097bfdc7-9d3d-852f-eba3-e7ea2e0c4239
* Content: [Migrating C++/CLI projects to .NET Core and .NET 5+ - .NET Core](https://learn.microsoft.com/en-us/dotnet/core/porting/cpp-cli)
* Content Source: [docs/core/porting/cpp-cli.md](https://github.com/dotnet/docs/blob/main/docs/core/porting/cpp-cli.md)
* Product: **dotnet-fundamentals**
* GitHub Login: @mjrousos
* Microsoft Alias: **dotnetcontent**

### PR DESCRIPTION
**Pull request by v-thepet**
## Summary

Describe your changes here.

Addresses but might not actually fix #33536 (?)

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/core/porting/cpp-cli.md](https://github.com/dotnet/docs/blob/2ed2ba013d34f8f56228f5641ae355aef528b1a0/docs/core/porting/cpp-cli.md) | [How to port a C++/CLI project to .NET Core or .NET 5+](https://review.learn.microsoft.com/en-us/dotnet/core/porting/cpp-cli?branch=pr-en-us-37687) |

### CODE REVIEW

File: docs/core/porting/cpp-cli.md [hunk 1]
```diff
@@ -13,79 +13,78 @@ Beginning with .NET Core 3.1 and Visual Studio 2019, [C++/CLI projects](/cpp/dot
...
+            "framework": {
+               "name": "Microsoft.NETCore.App",
+               "version": "3.1.0"
+            }
+         }
+   }
+   ```
+
+>[!NOTE]
+>C++/CLI assemblies can be loaded multiple times, each time into a new <xref:System.Runtime.Loader.AssemblyLoadContext>. If the first time managed code in a C++/CLI assembly is executed is from a native caller, the assembly is loaded into a separate `AssemblyLoadContext`. If the first time managed code is executed is from a managed caller, the assembly is loaded into the same `AssemblyLoadContext` as the caller, usually the default.
```
Comment (gewarren) [inline docs/core/porting/cpp-cli.md:86]:
```suggestion
>C++/CLI assemblies that target .NET 7 or a later version are always loaded into the default `AssemblyLoadContext`. However, in .NET 6 and earlier versions, C++/CLI assemblies might be loaded multiple times, each time into a new <xref:System.Runtime.Loader.AssemblyLoadContext>. If the first time that managed code in a C++/CLI assembly is executed:
>
> - Is from a native caller, the assembly is loaded into a separate `AssemblyLoadContext`.
> - Is from a managed caller, the assembly is loaded into the same `AssemblyLoadContext` as the caller, usually the default.
```

File: docs/core/porting/cpp-cli.md [hunk 2]
```diff
@@ -13,79 +13,78 @@ Beginning with .NET Core 3.1 and Visual Studio 2019, [C++/CLI projects](/cpp/dot
...
+>
+>To always load your C++/CLI assembly into the default `AssemblyLoadContext`, you can add an "initialize" style call from your entry-point assembly to your C++/CLI assembly. For more information, see this [dotnet/runtime issue](https://github.com/dotnet/runtime/issues/61105).
+>
```
Comment (gewarren) [inline docs/core/porting/cpp-cli.md:89]:
```suggestion
```

File: docs/core/porting/cpp-cli.md [hunk 3]
```diff
@@ -13,79 +13,78 @@ Beginning with .NET Core 3.1 and Visual Studio 2019, [C++/CLI projects](/cpp/dot
...
+>This limitation is removed starting in .NET 7. C++/CLI assemblies that target .NET 7 or a later version are always loaded into the default `AssemblyLoadContext`.
```
Comment (gewarren) [inline docs/core/porting/cpp-cli.md:90]:
```suggestion
```

File: docs/core/porting/cpp-cli.md [hunk 4]
```diff
@@ -13,79 +13,78 @@ Beginning with .NET Core 3.1 and Visual Studio 2019, [C++/CLI projects](/cpp/dot
...
+2. Replace `<TargetFrameworkVersion>` properties with `<TargetFramework>netcoreapp3.1</TargetFramework>`. Be sure to change the tag as well as the value.
+3. Remove any .NET Framework references to `System`, `System.Data`, `System.Windows.Forms`, and `System.Xml`, like `<Reference Include="System" />`. .NET Core SDK assemblies are automatically referenced when using `<CLRSupport>NetCore</CLRSupport>`.
 4. Update API usage in *.cpp* files, as necessary, to remove APIs unavailable to .NET Core. Because C++/CLI projects tend to be fairly thin interop layers, there are often not many changes needed. You can use the [.NET Portability Analyzer](../../standard/analyzers/portability-analyzer.md) to identify unsupported .NET APIs used by C++/CLI binaries just as with purely managed binaries.
 
 ### WPF and Windows Forms usage
 
+You can use an app with a native entry point that displays a Windows Forms form via C++/CLI. You can also migrate a solution with a managed entry point interoperating with native dependencies via C++/CLI.
```
Comment (gewarren) [inline docs/core/porting/cpp-cli.md:33]: I don't think I would add this since it's not entirely clear to me.

**Comment by adegeo** [comment]
To me, issue https://github.com/dotnet/docs/issues/33536 doesn't make sense.. I added a comment to the end of that issue stating this.

**Review by gewarren — APPROVED** [review]

**Comment by srikrishnagu** [comment]
This is very valid documentation issue this documentation has to be updated accordingly with appropriate details on what is the recommended way to get this done in this document https://learn.microsoft.com/en-us/dotnet/core/porting/cpp-cli

**Pull request merged**